### PR TITLE
[WIP] PR Check Enhancement

### PR DESCRIPTION
**Pull request by maknop**
## Jira Ticket

[RHCLOUD-24752](https://issues.redhat.com/browse/RHCLOUD-24752)

## Description

This change will convert the `pr_check.sh` to the `Jenkinsfile` script.

### CODE REVIEW

**Review by Victoremepunto — CHANGES_REQUESTED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -1,14 +1,223 @@
...
+        stage('Build test image') {
+            when {
+                expression {
+                    sh(script: "egrep 'lgtm|pr-check-build|*smoke-tests' ${LABELS_DIR}/github_labels.txt || true", returnStdout: true) != true
+                }
+            }
+            steps {
+                withVault([configuration: configuration, vaultSecrets: secrets]) {
+                    sh '''
+                        if egrep 'aws-smoke-tests' ${LABELS_DIR}/github_labels.txt &>/dev/null
```
Comment (Victoremepunto) [inline Jenkinsfile:110]: This sets `IQE_FILTER_EXPRESSION` and optionally `IQE_MARKER_EXPRESSION` env variables, to affect the behavior of the `Run Smoke Tests` Stage later on, but I don't think the environment variables remain set between stages.

**Review by Victoremepunto — APPROVED** [review]

**Review by maskarb — APPROVED** [review]

**Pull request merged**